src/utils/hooks/useTriggerProcessing.tsx

```diff
@@ -1,25 +1,62 @@
 import { useState, useEffect } from 'react'
 import { useGetTriggersQuery } from '../../utils/generated/graphql'
 
-type triggerTypes = 'applicationTrigger' | 'reviewTrigger'
+type TriggerType = 'applicationTrigger' | 'reviewAssignmentTrigger' | 'reviewTrigger'
 
-const useTriggerProcessing = (props: { serialNumber: string; trigger: triggerTypes }) => {
-  const { serialNumber, trigger } = props
+interface TriggerQueryProps {
+  serialNumber?: string
+  reviewAssignmentId?: number
+  reviewId?: number
+  triggerType?: TriggerType
+}
+
+const useTriggerProcessing = ({
+  serialNumber,
+  reviewAssignmentId,
+  reviewId,
+  triggerType,
+}: TriggerQueryProps) => {
   const [isProcessing, setIsProcessing] = useState(true)
   const [triggerError, setTriggerError] = useState(false)
 
+  // Ensure at least one the primary identifiers is provided
+  if (!serialNumber && !reviewAssignmentId && !reviewId) {
+    console.log('INVALID QUERY')
+    setIsProcessing(false)
+    setTriggerError(true)
+  }
+
+  // If triggerType not provided, infer it from the supplied ID
+  const inferredTriggerType = inferTriggerType(triggerType, serialNumber, reviewAssignmentId)
+
   const { data, loading, error } = useGetTriggersQuery({
     variables: {
       serial: serialNumber,
+      reviewAssignmentId,
+      reviewId,
     },
     pollInterval: 500,
     skip: !isProcessing,
     fetchPolicy: 'no-cache',
   })
 
   useEffect((): any => {
-    if (data?.applicationTriggerStates?.nodes[0]) {
-      const triggerRequested = data?.applicationTriggerStates?.nodes[0][trigger]
+    if (data?.applicationTriggerStates?.nodes?.[0]) {
+      let triggerRequested
+      switch (inferredTriggerType) {
+        case 'applicationTrigger':
+          triggerRequested = data?.applicationTriggerStates?.nodes[0]?.applicationTrigger
+          break
+        case 'reviewAssignmentTrigger':
+          triggerRequested = data?.applicationTriggerStates?.nodes[0]?.reviewAssignmentTrigger
+          break
+        case 'reviewTrigger':
+          triggerRequested = data?.applicationTriggerStates?.nodes[0]?.reviewTrigger
```

Comment:
Also better to set this `data?.applicationTriggerStates?.nodes[0]?` to a variable to make the code cleaner

Reply:
Okay, assigned the whole long thing to `triggers` a63c0e5